src/main/groovy/org/checkerframework/gradle/plugin/CheckerFrameworkPlugin.groovy

```diff
@@ -142,6 +145,12 @@ final class CheckerFrameworkPlugin implements Plugin<Project> {
       project.tasks.withType(AbstractCompile).all { compile ->
         if (compile.hasProperty('options') && (!userConfig.excludeTests || !compile.name.toLowerCase().contains("test"))) {
           compile.options.annotationProcessorPath = project.configurations.checkerFramework
+          if (javaVersion.java8 && new JarFile(project.configurations.checkerFramework.asPath)
+                          .getManifest().getMainAttributes().getValue('Implementation-Version') >= "3") {
```

Comment:
I just realized that `.getManifest().getMainAttributes().getValue("Implementation-Version").tokenize(".")[0].toInteger() >= 3` might be a better way to compare the version. I've tested this change and it can be copied directly.

Then I think we're good. Thanks for you help!

Reply:
The other alternative would be to just test against the `cfVersion` in the user configuration for the Gradle plugin. Is there any compelling case where that wouldn't be set but we would want to use the ErrorProne compiler?

Reply:
The reason for reading `Implementation-Version` is that sometimes we may need to use a locally built checker-framework whose version can only be decided by examining its `checker.jar`.

As for using the errorprone compiler, it's because it contains classes of Java 9 that are dependent by checker-framework with version >=3. @wmdietl

Reply:
Okay, that makes sense.

Reply:
If there is a configuration variable that contains the CF version to use, that sounds fine to me. If the user mis-configures the project and uses a 3.0 version without updating the version in the configuration, they will run into errors.
Especially if unzipping checker.jar is causing problems, I'm for going for a simpler route.